keps/sig-api-machinery/3488-cel-admission-control/README.md

```diff
@@ -1061,16 +1062,37 @@ Details to consider:
 
 #### CEL Expression Composition
 
+##### Use Cases
+
+###### Reusing/memoizing an expensive computation
+
+For a CEL expression that runs in O(n^2) time or worse (or otherwise
+takes a significant amount of time to execute), it would be nice to only run
+it when necessary. For instance, if multiple validation expressions used the
+same expensive subexpression, that subexpression could be refactored out.
+
+###### Code re-use for complicated subexpressions
+
+An subexpression may not be computationally expensive, but could still be
+intricate enough that copy-pasting it would be considered a code smell.
```

Comment:
I consider this the primary motivation for this feature.

Spell out what is wrong with copy-pasting here? Redundancy? Higher maintenance cost? Potential for code drift?

Reply:
Right, I've re-ordered the use case section and made the part re: copy-pasting a bit more clear.